src/brod_group_subscriber.erl

```diff
@@ -543,23 +568,31 @@ subscribe_partitions(#state{ client    = Client
 subscribe_partition(Client, Consumer) ->
   #consumer{ topic_partition = {Topic, Partition}
            , consumer_pid    = Pid
-           , begin_offset    = BeginOffset0
+           , begin_offset    = BeginOffset
            , acked_offset    = AckedOffset
            } = Consumer,
   case brod_utils:is_pid_alive(Pid) of
     true ->
       Consumer;
     false ->
-      %% fetch from the last acked offset + 1
-      %% otherwise fetch from the assigned begin_offset
-      BeginOffset = case AckedOffset of
-                      ?undef        -> BeginOffset0;
-                      N when N >= 0 -> N + 1
-                    end,
       Options =
-        case BeginOffset =:= ?undef of
-          true  -> []; %% fetch from 'begin_offset' in consumer config
-          false -> [{begin_offset, BeginOffset}]
+        if BeginOffset =:= ?undef andalso
```

Comment:
The logic in this block looks duplicated elsewhere and might warrant being factored out into a helper function.